Xamarin.Forms.Core/Storyboard.cs

```diff
@@ -0,0 +1,551 @@
+				{
+					item.SetValue(BindingContextProperty, (newValue as VisualElement).BindingContext);
+				}
+			});
+
+		public static readonly BindableProperty TargetPropertyProperty = BindableProperty.CreateAttached("TargetProperty", typeof(BindableProperty), typeof(Timeline), null);
+
+		public static void SetTarget(BindableObject bindable, VisualElement value)
+		{
+			bindable.SetValue(TargetNameProperty, value);
+			SetInheritedBindingContext(bindable, value?.BindingContext);
+		}
+
+		public static void SetTargetName(BindableObject bindable, string value)
+		{
+			bindable.SetValue(TargetNameProperty, value);
+		}
+
+		public static VisualElement GetTargetName(BindableObject bindable)
+		{
+			return (VisualElement)bindable.GetValue(TargetNameProperty);
+		}
+
+		public static VisualElement GetTarget(BindableObject bindable)
+		{
+			return (VisualElement)bindable.GetValue(TargetNameProperty);
+		}
+
+		public static void SetTargetProperty(BindableObject bindable, BindableProperty value)
+		{
+			bindable.SetValue(TargetPropertyProperty, value);
+		}
+
+		protected override void OnBindingContextChanged()
+		{
+			base.OnBindingContextChanged();
+			foreach (var item in Children)
+			{
+				item.SetValue(BindingContextProperty, BindingContext);
+				//SetInheritedBindingContext(item, BindingContext);
+
+			}
+		}
+
+		public static BindableProperty GetTargetProperty(BindableObject bindable)
+		{
+			return (BindableProperty)bindable.GetValue(TargetPropertyProperty);
+		}
+
+		public Storyboard()
+		{
+			_timelines = new List<Timeline>();
+		}
+
+		public void Begin(Timeline item)
+		{
+			_animationsPerElement.Clear();
+			_runningAnimations.Clear();
+			var maxDuration = Duration;
+			var defaultTarget = Storyboard.GetTarget(this);
+			var defaultTargetProperty = Storyboard.GetTargetProperty(this);
+
+
+			if (item is Animation<Color> timeline)
+			{
+				maxDuration = AddAnimationColor(maxDuration, defaultTarget, defaultTargetProperty, timeline);
+			}
+
+			if (item is DoubleAnimation doubleAnimation)
+			{
+				AddDoubleAnimation(defaultTarget, defaultTargetProperty, doubleAnimation);
+			}
+
+			if (item is PositionAnimation positionAnimation)
+			{
+				AddPositionAnimation(defaultTarget, defaultTargetProperty, positionAnimation);
+			}
+			foreach (var anim in _animationsPerElement)
+			{
+				var elementAnimation = new Animation();
+				foreach (var animation in anim.Value)
+				{
+					elementAnimation.Add(0, 1, animation);
+
+				}
+				_runningAnimations.Add(elementAnimation);
+				elementAnimation.Commit(anim.Key, "ChildAnimations", 16, maxDuration, finished: (v, c) => Completed?.Invoke(this, new EventArgs()));
+			}
+		}
+
+		public new void Begin()
+		{
+			_animationsPerElement.Clear();
+			_runningAnimations.Clear();
+			var maxDuration = Duration;
+			var defaultTarget = Storyboard.GetTarget(this);
+			var defaultTargetProperty = Storyboard.GetTargetProperty(this);
+			foreach (var item in Children)
```

Comment:
The whole logic here seems duplicate of Begin(Timeline item)`, or maybe I am wrong? Why not call `Begin(item)` ?

Reply:
Yeah this could be refactor better, but the idea is when 1 property change you call Begin on that animation , if you Begin a Storyboard it begins all animations. But sure it could be refactored